### PR TITLE
docs: Add versioning file

### PR DESCRIPTION
**Pull request by seebees**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

# Check any applicable:
- [ ] Were any files moved? Moving files changes their URL, which breaks all hyperlinks to the files.

### CODE REVIEW

File: VERSIONING.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+## Patch versions
+
+Patch versions changes are meant only for bug fixes,
+and will not break compatibility of the current major version.
+A patch release will contain a collection of minor bug fixes, 
+or individual major and security bug fixes, depending on severity.
+A patch release will also include warning of upcoming breaking changes, whenever possible.
+
+What this means for you
+=======================
```
Comment (mattsb42-aws) [inline VERSIONING.md]: Needs markdown-style heading notation.

File: VERSIONING.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+## Major versions
+
+Major version changes are significant and expected to break backwards compatibility.
+
+## Minor versions
+
+Minor version changes will not break compatibility between the previous minor versions;
+to do so is a bug.
+Encryption SDK changes will also involve addition of optional features, and non-breaking enhancements.
+Additionally any change to the version of a dependency.
```
Comment (lizroth) [inline VERSIONING.md]: "Additionally, any change to the version of a dependency is a minor version change."

or

"Minor version changes additionally include any change to the version of a dependency."

I don't think that's a sentence as is. :) Just trying to minimize grok effort for readers.

File: VERSIONING.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  * If you have good unit and integration tests, these changes are generally safe to pick up automatically.
+
+* Z changes will not require any changes to your code. Z changes are intended to be picked up automatically.
+
+  * Good unit and integration tests are always recommended.
+
+# Semantic Commits
+
+We seek to increase clarity at all levels of the update and releases process.
+We require pull requests adhere to the [Conventional Commits][conventional-commits] spec,
```
Comment (lizroth) [inline VERSIONING.md]: Have we looked at / tried out any of the automated enforcement tools? What would it take to get some of those running, do we know?

**Comment by SalusaSecondus** [comment]
1. Conventional commits seems like a neat idea and depending on how it works here may be worth rolling out more widely
2. The other ESDKs don't follow semantic versioning and so if this does, it will be different. See the [java VERSIONING file](https://github.com/aws/aws-encryption-sdk-java/blob/master/VERSIONING.rst) as an example. Minor changes are permitted backwards incompatible changes which can be:
  a. Usually be ignored because they are rarely encountered/edge-case, and
  b. Can be automatically picked up by any system with a reasonable set of unit-/integration-tests, and
  c. (If not extremely urgent) are warned with deprecation notices in a prior Minor/Patch release

File: VERSIONING.md (hunk 3 above)
Comment (seebees) [reply VERSIONING.md]: I am working on that.  exploring https://github.com/probot/semantic-pull-requests

Comment (lizroth) [reply VERSIONING.md]: Love it!

File: VERSIONING.md [hunk 4]
```diff
@@ -11,28 +11,14 @@ Major version changes are significant and expected to break backwards compatibil
...
-We recommend running the most recent version. Here are our suggestions for managing updates:
-
-* X changes will require some effort to incorporate.
-* Y changes will not require significant effort to incorporate.
-
-  * If you have good unit and integration tests, these changes are generally safe to pick up automatically.
-
-* Z changes will not require any changes to your code. Z changes are intended to be picked up automatically.
-
-  * Good unit and integration tests are always recommended.
```
Comment (lizroth) [inline VERSIONING.md]: I think we could have left these as expressions of our intent (especially the bit about if we know we will need to make breaking changes soon, we will make an effort to warn you), but I don't think removing them is a problem either, given that we're stating our use of CC/SemVer.

**Comment by SalusaSecondus** [comment]
To be clear. My concerns aside, I have no objections to us moving forward with this proposed versioning scheme.
